apps/shopify/src/skuResolvers.js

```diff
@@ -48,11 +48,14 @@ export const fetchCollectionPreviews = async (skus, config) => {
 
   const response = await shopifyClient.collection.fetchAll(SHOPIFY_ENTITY_LIMIT);
   if (response.length > 0) {
-    while (response[response.length - 1].hasNextPage) {
+    let hasNextPage = response.length === SHOPIFY_ENTITY_LIMIT;
+    while (hasNextPage && response[response.length - 1].hasNextPage) {
```

Comment:
If `response` is an array (which I think should be renamed to something more descriptive), why don't we just `.map` this and return the array of promises. And then resolve those promises. Didn't you do this promise/resolution pattern somewhere as well in this file?

Reply:
Like you did here https://github.com/contentful/apps/blob/master/apps/shopify/src/skuResolvers.js#L89-L94

Reply:
This was changed significantly in the current commit. Mind taking a look?